ggml/src/ggml-vulkan/vulkan-shaders/mul_mm.comp

```diff
@@ -65,13 +72,26 @@ layout (constant_id = 5) const uint WN = 32;
 layout (constant_id = 6) const uint WMITER = 2;
 layout (constant_id = 7) const uint TM = 4;
 layout (constant_id = 8) const uint TN = 2;
-layout (constant_id = 9) const uint WARP = 32;
+layout (constant_id = 9) const uint TK = 1;  // Only needed for coopmat
+layout (constant_id = 10) const uint WARP = 32;
+
+#ifdef COOPMAT
+#define SHMEM_STRIDE (BK + 8)
+#else
+#define SHMEM_STRIDE (BK + 1)
+#endif
 
-shared FLOAT_TYPE buf_a[BM * (BK+1)];
-shared FLOAT_TYPE buf_b[BN * (BK+1)];
+shared FLOAT_TYPE buf_a[BM * SHMEM_STRIDE];
+shared FLOAT_TYPE buf_b[BN * SHMEM_STRIDE];
 
 #ifdef MUL_MAT_ID
 shared u16vec2 row_ids[3072];
+#endif // MUL_MAT_ID
+
+#define NUM_WARPS (BLOCK_SIZE / WARP)
```

Comment:
I wonder if this requires subgroup_size_control for Intel.

Reply:
I'm OK with not addressing this for now, since Intel support is currently disabled. But it would be interesting to try as a followup.

Reply:
It would be interesting, but it's not gonna fix an unfinished coopmat implementation on the driver side. I'll implement the subgroup size extension in another PR, it might be good for AMD RDNA as well, otherwise we can't be sure which subgroup sizes we'll get.